open_learning_ai_tutor/message_tutor.py

```diff
@@ -0,0 +1,46 @@
+from open_learning_ai_tutor.tutor import Tutor
+from open_learning_ai_tutor.prompts import get_assessment_prompt, get_tutor_prompt
+from open_learning_ai_tutor.utils import filter_out_system_messages
+from open_learning_ai_tutor.intent_selector import get_intent
+from open_learning_ai_tutor.constants import Intent
+
+
+def message_tutor(
+    problem: str,
+    problem_set: str,
+    client,
+    new_messages,
+    chat_history,
+    assessment_history,
+    intent_history,
+    tools,
+):
+    tutor = Tutor(
+        client,
+        tools=tools,
+    )
+    assessment_prompt = get_assessment_prompt(
+        problem, problem_set, assessment_history, new_messages
+    )
+    assessment_response = tutor.get_response(assessment_prompt)
+    new_assessment_history = assessment_response["messages"]
+
+    previous_intent = intent_history[-1] if intent_history else [Intent.S_STRATEGY]
+    new_intent = get_intent(new_assessment_history[-1].content, previous_intent)
+
+    prompt = get_tutor_prompt(
+        problem,
+        problem_set,
+        chat_history,
+        new_intent,
+    )
+
+    response = tutor.get_response(prompt)
+    new_history = filter_out_system_messages(response["messages"])
+    new_assessment_history = new_assessment_history[
```

Comment:
consider adding some checks around length of new_assessment_history (and logging errors if it doesnt contain expected elements)